software/firmware/europi_config.py

```diff
@@ -6,20 +6,85 @@
 PICO_DEFAULT_CPU_FREQ = 125_000_000
 OVERCLOCKED_CPU_FREQ = 250_000_000
 
+# Moog/Eurorack standard is 1.0 volts per octave
+MOOG_VOLTS_PER_OCTAVE = 1.0
+
+# Buchla standard is 1.2 volts per octave (0.1 volts per semitone)
+BUCHLA_VOLTS_PER_OCTAVE = 1.2
+
 
 class EuroPiConfig:
-    """This class provides EuroPi's global config points."""
+    """This class provides EuroPi's global config points.
+
+    To override the default values, create /config/config_EuroPiConfig.json on the Raspberry Pi Pico
+    and populate it with a JSON object. e.g. if your build has the oled mounted upside-down compared
+    to normal, the contents of /config/config_EuroPiConfig.json should look like this:
+
+    {
+        "rotate_display": true
+    }
+    """
 
     @classmethod
     def config_points(cls):
+        # fmt: off
         return [
-            configuration.choice(name="pico_model", choices=["pico", "pico w"], default="pico"),
+            # EuroPi revision -- this is currently unused, but reserved for future expansion
+            configuration.choice(
+                name="europi_model",
+                choices = ["europi", "europi_lc", "europi_x"],
+                default="europi"
+            ),
+
+            # CPU & board settings
+            configuration.choice(
+                name="pico_model",
+                choices=["pico", "pico w"],
+                default="pico"
+            ),
             configuration.choice(
                 name="cpu_freq",
                 choices=[PICO_DEFAULT_CPU_FREQ, OVERCLOCKED_CPU_FREQ],
                 default=OVERCLOCKED_CPU_FREQ,
             ),
+
+            # Display settings
+            configuration.choice(
+                name="rotate_display",
+                choices=[False, True],
+                default=False
+            ),
+            configuration.integer(
+                name="display_width",
```

Comment:
I'm torn between both of your arguments. I do completely agree that people might use different sized OLEDs with any model, but I also know that a lot of people will want their module/variants of the module to be usable without having to know and manually set those values.

I think that if we set in stone that modules will use GPIOs pulled high and low to determine the model, then a single button based on model is unnecessary because this will happen on boot (maybe a reset function would be good though to set them all to what they *should* be based on GPIO model setting)

If we aren't setting that in stone, then I think a solution could be made where you can set those values individually, but you can also scroll down and click
```
Set all values depending on model:
EuroPi-X
EuroPi-LC
```

This solution relies on the config menu being a menu accessible interactive script though, which it isn't yet. For now my instinct is to allow them to be changed as this PR does, and chances are the config menu as a script work will happen before the X becomes consumer available. Until then there's no harm in having them changeable but with a default value of 128 and 32 so normal users don't have to worry about it

Reply:
I gave it a try with a 120px width and it worked alright in the few scripts that I tried. So I agree that there's not much difference in making these settings slightly more accessible. 

We can work out the details on how settings for different models will work in a future PR.